examples/platform_channel/android/app/src/main/java/com/example/platformchannel/MainActivity.java

```diff
@@ -45,6 +69,23 @@ public void onMethodCall(MethodCall call, Response response) {
     );
   }
 
+  private BroadcastReceiver createChargingStateChangeReceiver(final EventSink events) {
+    return new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        int status = intent.getIntExtra(BatteryManager.EXTRA_STATUS, -1);
+
+        if (status == BatteryManager.BATTERY_STATUS_UNKNOWN) {
```

Comment:
In my opinion no. Since there are only two cases I think this is simpler. With the switch it also takes op two lines more. But if you feel strongly about it I don't mind changing it.